includes/class-wc-stripe-intent-controller.php

```diff
@@ -954,4 +960,25 @@ public function create_and_confirm_setup_intent_ajax() {
 	private function request_needs_redirection( $payment_methods ) {
 		return 1 === count( $payment_methods ) && 'card' !== $payment_methods[0];
 	}
+
+	/**
+	 * Searches for a compatible payment intent for the given order and payment method types.
+	 *
+	 * @param int $customer The customer ID.
+	 * @param int $order_id The order ID.
+	 * @param array $payment_method_types The payment method types.
+	 * @return object|null
+	 * @throws WC_Stripe_Exception
+	 */
+	private function query_for_compatible_payment_intent( $customer, $order_id, $payment_method_types ) {
+		$query_string   = sprintf( "customer:'%s' AND status:'requires_payment_method' AND metadata['order_id']:'%s'", $customer, $order_id );
+		$search_results = WC_Stripe_API::request( [ 'query' => $query_string ], 'payment_intents/search', 'GET' );
```

Comment:
Did we evaluate using the payment intent that's already associated with the WC Order, instead of searching for an old one in Stripe?

When processing the order, we store the payment intent ID as a WC Order meta:

- [save_intent_to_order()](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/deferred-intent/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L1532-L1557)
- [add_payment_intent_to_order()](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/deferred-intent/includes/class-wc-stripe-helper.php#L766-L790)

If there's an associated payment intent, we could:
- Get the ID from the order meta
- [Retrieve the payment intent object](https://stripe.com/docs/api/payment_intents/retrieve) from Stripe by its ID
- Confirm whether we could reuse it, given the new payment attempt
- And [update it](https://stripe.com/docs/api/payment_intents/update) with the new payment information

If there's no payment intent associated with the WC Order, we'd just create a new one. No need to search for an existing one in Stripe.

Reply:
Oh, it makes much more sense! I replaced the existing logic with your suggestion. Thank you